mapie/classification.py

```diff
@@ -601,16 +606,18 @@ def _fit_and_predict_oof_model(
           of shape (n_samples_val,).
 
         """
-        X_train, y_train, X_val, y_val = (
-            X[train_index], y[train_index], X[val_index], y[val_index]
-        )
+        X_train = _safe_indexing(X, train_index)
```

Comment:
This function allows you to keep the type of the array (list, numpy or pandas), right ?

Reply:
Exactly. The bug was partly due to the conversion of pandas arrays to numpy arrays, loosing column names required in some ColumnTransformer of scikit-learn.